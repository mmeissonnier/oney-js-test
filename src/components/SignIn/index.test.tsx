import React from 'react';
import { render } from '@testing-library/react';
import { screen, fireEvent } from '@testing-library/dom';
import { SignIn } from '.';
import { MemoryRouter } from 'react-router-dom';

describe('components/SignIn', () => {
  it('should render correctly', () => {
    const { container } = render(
      <MemoryRouter>
        <SignIn onSubmit={jest.fn()} />
      </MemoryRouter>
    );
    expect(container).toMatchSnapshot();
  });

  it('should call onSubmit with correct values', () => {
    const handleSubmit = jest.fn();
    render(
      <MemoryRouter>
        <SignIn onSubmit={handleSubmit} />
      </MemoryRouter>
    );
    fireEvent.change(screen.getByTestId('input.email'), {
      target: { value: 'foo@test.com' },
    });
    fireEvent.change(screen.getByTestId('input.name'), {
      target: { value: 'foo' },
    });
    fireEvent.click(screen.getByText('Create my account'));
    expect(handleSubmit).toHaveBeenCalledWith({
      email: 'foo@test.com',
      name: 'foo',
    });
  });

  it('should display an error when email is invalid', () => {
    const handleSubmit = jest.fn();
    render(
      <MemoryRouter>
        <SignIn onSubmit={handleSubmit} />
      </MemoryRouter>
    );
    fireEvent.change(screen.getByTestId('input.email'), {
      target: { value: 'foo-bar-biz' },
    });
    fireEvent.click(screen.getByText('Create my account'));
    expect(handleSubmit).not.toHaveBeenCalledWith();
    expect(screen.getByText('Invalid email')).toBeDefined();
  });
});
