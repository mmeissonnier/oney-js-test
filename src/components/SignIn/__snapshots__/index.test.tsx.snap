// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/SignIn should render correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  margin: 0;
  color: white;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c2 {
  font-weight: bold;
  color: red;
}

<div>
  <div
    class="c0"
  >
    <form
      class="c1"
    >
      <label
        for="email"
      >
        Email
      </label>
      <input
        data-testid="input.email"
        id="email"
        name="email"
        type="text"
        value=""
      />
      <label
        for="name"
      >
        Name
      </label>
      <input
        data-testid="input.name"
        id="name"
        name="name"
        type="text"
        value=""
      />
      <button
        data-testid="signin.form.submit"
        type="submit"
      >
        Create my account
      </button>
      <p
        class="c2"
      />
    </form>
  </div>
</div>
`;
